Source/Managers/DFU/FirmwareUpgradeManager.swift

```diff
@@ -57,18 +57,34 @@ public class FirmwareUpgradeManager : FirmwareUpgradeController, ConnectionObser
     
     /// Start the firmware upgrade.
     public func start(data: Data) throws {
+        try start(images: [(0, data)])
+    }
+    
+    public func start(images: [(Int, Data)]) throws {
         objc_sync_enter(self)
         if state != .none {
             log(msg: "Firmware upgrade is already in progress", atLevel: .warning)
             return
         }
-        imageData = data
-        hash = try McuMgrImage(data: imageData).hash
+        
+        i = 0
+        self.images = try images.map {
+            try FirmwareUpgradeImage(slot: $0.0, image: $0.1)
+        }
+        
+        if self.images.count > 1, mode != .testOnly {
+            // As of this writing, multi-image does not support testing. We can only
```

Comment:
Don't hard code this. It should be possible to select any mode, some may work better or worse, but customers can improve their implementation on the firmware side.

Reply:
Moreover, the preferred mode should be `.confirmOnly`.